include/boost/math/tools/polynomial.hpp

```diff
@@ -527,64 +531,59 @@ class polynomial :
         return *this;
     }
 
+    template <class U>
+    polynomial& modulus(const U& value)
+    {
+       typedef typename std::vector<T>::iterator iter;
+       for (iter it = m_data.begin(); it != m_data.end(); ++it)
+           *it -= T(value * T(*it / value));
+       return *this;
+    }
```

Comment:
Yes, `operator %=(U)`, and the other self-modifying operators for scalar arguments, are all still templated. We can change that if you want to be more strict.